include/deal.II/base/parameter_handler.h

```diff
@@ -1700,6 +1727,16 @@ class ParameterHandler : public Subscriptor
             const unsigned int current_line_n,
             const bool         skip_undefined);
 
+  /**
+   * Sort all parameters of the subsection given by the
+   * @p target_subsection_path argument in alphabetical order,
+   * as well as all subsections within it recursively.
+   */
+  void
+  recursively_sort_parameters(
+    boost::property_tree::ptree &   tree,
+    const std::vector<std::string> &target_subsection_path) const;
```

Comment:
We typically have the input arguments first, then the output arguments. Would you mind reversing the order of the two arguments? 

You make it sound in the description of this function as if the function *only* operates on the given arguments. Could it then be made `static` (and then of course non-`const`)?

Reply:
It can also be moved into an anonymous namespace in the .cc instead, but I would suggest to move forward with the PR instead (after we figure out if the functionality is correct).

Reply:
The order of the two arguments has been reversed, and the method has been made static.

Currently it is not possible to move it into the anonymous namespace since it makes use of `collate_path_string` which is non-static and in turn refers to `path_separator`, which is a *private* static class member. This of course can be easily fixed, but since this goes beyond the scope of this PR I preferred not to implement here also these changes.